src/AzureAuth/Program.cs

```diff
@@ -35,14 +36,26 @@ private static void Main(string[] args)
             TelemetryOutput backend = TelemetryOutput.StandardOut;
 
             // We will only send telemetry if we are given an ingestion token via
-            // environment variable. We *have* to do this here, rather than in
+            // environment variable or registry. We *have* to do this here, rather than in
             // `CommandMain.OnExecute`, because Lasso doesn't have a way of allowing a
             // `CommandLineApplication` to dynamically set telemetry configuration. Even if
             // it did, we can't guarantee that a failure in command line parsing wouldn't
             // trigger telemetry before we ever get to disable it.
             //
             // To disable telemetry a user need only leave this environment variable unset. It's off by default.
             string applicationInsightsIngestionToken = Environment.GetEnvironmentVariable(EnvVars.ApplicationInsightsIngestionTokenEnvVar);
+
+            if (string.IsNullOrEmpty(applicationInsightsIngestionToken))
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))

```

Comment:
Because this is Windows only, I think we should use `#ifdef WINDOWS` instead of a runtime check. We don't need to both checking on other platforms, and in the future, we may have a different block to conditionally compile in the else case for Mac, and for Linux. I don't feel strongly about this, but I think it would be good to know ho many `ms` it actually takes to check the OS at runtime. I've seen that code memoized in many other codebases to avoid doing it more than once, which makes me worry it's longer than one might think.

Reply:
If we use `if PlatformWindows`, we have to suppress CA1416.

#if PlatformWindows
#pragma warning disable CA1416 // Validate platform compatibility
                applicationInsightsIngestionToken = Win32.Registry.GetValue(
                    EnvVars.ApplicationInsightsIngestionTokenRegKeyPath,
                    EnvVars.ApplicationInsightsIngestionTokenRegKeyName,
                    null) as string;
#pragma warning restore CA1416 // Validate platform compatibility
#endif

Reply:
Does this more properly may require more changes to our targeting configuration, so I'm fine staying with the runtime for now.